lib/components/widgets/scroll_direction_listener.dart

```diff
@@ -64,16 +79,31 @@ class _ScrollDirectionListenerState
     super.dispose();
   }
 
-  bool _onNotification(UserScrollNotification notification) {
+  bool _onNotification(ScrollNotification notification) {
     if (widget.depth != null && notification.depth != widget.depth) {
       return false;
     }
 
-    if (mounted &&
-        notification.direction != ScrollDirection.idle &&
-        notification.metrics.maxScrollExtent > _maxScrollExtentThreshold &&
-        _direction != notification.direction) {
-      setState(() => _direction = notification.direction);
+    final metrics = notification.metrics;
+
+    if (mounted && metrics.maxScrollExtent > _maxScrollExtentThreshold) {
+      final scrollValue = metrics.pixels;
+      final scrollDirection = scrollValue - _scrollValue < 0
```

Comment:
```suggestion
      final scrollDirection = scrollValue == 0 || scrollValue - _scrollValue < 0
```

While testing this out, I was able to get into a state where I was at the top but the scroll direction was still reporting a `ScrollDirection.reverse` and thus the tab bar was not showing.
This change prevents that.